app/src/main/java/nl/nfi/cellscanner/AppUpgradeReceiver.java

```diff
@@ -0,0 +1,31 @@
+package nl.nfi.cellscanner;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.widget.Toast;
+
+import nl.nfi.cellscanner.recorder.PermissionSupport;
+import nl.nfi.cellscanner.recorder.RecorderUtils;
+
+/**
+ * Responsible for starting the App after device boot
+ *
+ * Checks if the application is was left in a recording state before the reboot, if true
+ * starts the application after the boot sequence has been completed
+ */
+public class AppUpgradeReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        /*
+          Call the start of the application if the boot-up state of the application
+          is inRecordingState
+         */
+        if (RecorderUtils.inRecordingState(context)){
+            if (PermissionSupport.hasAccessCourseLocationPermission(context)) RecorderUtils.startService(context);
+            else RecorderUtils.setRecordingState(context, false);
```

Comment:
Deze setting `RecordingState` is iets dat de gebruiker opgeeft, dus een voorkeur van de gebruiker. Ik vind dat we die niet automatisch terug kunnen zetten zonder dat de gebruiker dat weet. De service kunnen we starten als de gebruiker dat wil. De permission dan checken binnen de service zelf.

Reply:
Dit gebeurt alleen wanneer de gebruiker de permissie heeft ingetrokken. De applicatie kan dan niet meer in een running state staan en moet dus wel van de recording sate afgehaald worden

Reply:
Dat hoeft niet; in de nieuwe android kan je ook toestemming geven voor de huidige sessie bijvoorbeeld. Kan unintended zijn. Plus, als de gebruiker zichzelf tegenspreekt kunnen we hem attenderen.

Reply:
We kunnen de gebruiker nogmaals een melding geven dat de app het nodig heeft. Dit betekend dat we een bericht moeten gaan sturen dat de app niet langer draaid. 
Wanneer we binnen de service een check doen, dan nemen we niet op terwijl de app wel in de record status staat, dat lijkt me kwalijker.

Reply:
Zoals bestproken, nieuwe ticket om dit proces te verbeteren